libs/platform/src/lib/components/form/select/commons/base-select.ts

```diff
@@ -0,0 +1,759 @@
+import {
+    AfterViewInit,
+    ChangeDetectorRef,
+    ContentChildren,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Host,
+    Input,
+    OnDestroy,
+    Optional,
+    Output,
+    QueryList,
+    Self,
+    SkipSelf,
+    TemplateRef,
+    ViewChild
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+import {
+    CONTROL,
+    DOWN_ARROW,
+    ENTER,
+    ESCAPE,
+    LEFT_ARROW,
+    RIGHT_ARROW,
+    SHIFT,
+    TAB,
+    UP_ARROW,
+    BACKSPACE
+} from '@angular/cdk/keycodes';
+
+import { fromEvent, isObservable, Observable, Subject, Subscription } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import {
+    FocusEscapeDirection,
+    KeyUtil,
+    ListComponent,
+    MobileModeConfig,
+    TemplateDirective,
+    PopoverFillMode
+} from '@fundamental-ngx/core';
+import {
+    ArraySelectDataSource,
+    SelectDataSource,
+    isDataSource,
+    isOptionItem,
+    MatchingBy,
+    ObservableSelectDataSource,
+    OptionItem
+} from '../../../../domain';
+
+import { isFunction, isJsObject, isString } from '../../../../utils/lang';
+import { CollectionBaseInput } from '../../collection-base.input';
+import { SelectComponent } from '../select/select.component';
+import { SelectConfig, MatchingStrategy } from '../select.config';
+import { ContentDensity, FormFieldControl } from '../../form-control';
+import { FormField } from '../../form-field';
+import { TextAlignment } from '../../combobox';
+
+export type FdpSelectDataSource<T> = SelectDataSource<T> | Observable<T[]> | T[];
+
+export class FdpSelectionChangeEvent {
+    constructor(
+        public source: SelectComponent,
+        public payload: any // Contains selected item
+    ) {}
+}
+
+@Directive()
+export abstract class BaseSelect extends CollectionBaseInput implements AfterViewInit, OnDestroy {
+    /** Provides maximum height for the optionPanel */
+    @Input()
+    maxHeight = '250px';
+
+    /**
+     * Todo: Name of the entity for which DataProvider will be loaded. You can either pass list of
+     * items or use this entityClass and internally we should be able to do lookup to some registry
+     * and retrieve the best matching DataProvider that is set on application level
+     */
+    @Input()
+    entityClass: string;
```

Comment:
EntityClass is only valid for datasources . We dont have usecase for DS yet for this component